### PR TITLE
Release 0.1.0

### PR DESCRIPTION
**Pull request by forevermatt**
Note: Since this is still a `0.x` release, anything can still change. That also means that I won't bother with data structure migrations yet.

### CODE REVIEW

**Review by longrunningprocess — COMMENTED** [review]
yay, getting pumped!

File: js/budget-app.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      'opts': {
+        'transaction': this.income
+      }
+    }
+  };
+  route(this.route.bind(this));
+};
+
+bb.BudgetApp.prototype.route = function(page, subPage, id) {
+  if ( ! page) {
```
Comment (longrunningprocess) [inline js/budget-app.js:113]: was the space before the bang intentional?

File: js/budget-app.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  var routeData = this.routes[path];
+  if (routeData != undefined) {
+    routeData.opts = routeData.opts || {};
+    routeData.opts.id = id;
+    routeData.opts.dm = this.dataManager;
+    try {
+      this.page.showTag(routeData.tagName, routeData.opts);
+    } catch (e) {
+      console.log(e);
```
Comment (longrunningprocess) [inline js/budget-app.js:132]: you might consider `console.error` for semantics.

File: js/data-manager.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+bb.DataManager.prototype.getTransactionById = function(id) {
+  return this.transactionService.getById(id);
+};
+
+bb.DataManager.prototype.getTransactions = function() {
+  return this.transactionService.getAll();
+};
+
+bb.DataManager.prototype.getTransactionsMatching = function(query) {
+  if (query === '' || query == undefined) {
```
Comment (longrunningprocess) [inline js/data-manager.js:82]: this could be `if (! query) return []` unless you are specifically allowing `null` through.

**Review by longrunningprocess — COMMENTED** [review]
Looks like you're having lots of fun!!!  Just curious if you're avoiding es6 sugar for backwards support or not, but it looked like there were plenty of places your code could shrink if you went with some of the new stuff.  

It seems riot is legit, would you say at this point?

File: js/budget-app.js (hunk 1 above)
Comment (forevermatt) [reply js/budget-app.js:113]: Kind of. :-) I could probably go back to using `if (!page) {` or `if (! page) {`. Do you see a particular benefit of using one or the other of those two options?

File: js/budget-app.js (hunk 2 above)
Comment (forevermatt) [reply js/budget-app.js:132]: Ahh, good catch.

File: js/budget-app.js (hunk 1 above)
Comment (longrunningprocess) [reply js/budget-app.js:113]: only personal preference :-)

**Pull request merged**